airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -83,9 +92,8 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Ite
             days_of_data_storage=self.days_of_data_storage,
         )
 
-    @staticmethod
-    def get_date_params(stream_slice: Mapping[str, Any], cursor_field: str, end_date: pendulum.datetime = None, time_unit: str = "months"):
-        end_date = end_date or pendulum.yesterday()
```

Comment:
I think there were some reasons to do it.

I suspect that Google Ads do not generate reports in real time. 

As far as I remember google need some time (a day) to have full report for given day.

If we set end_day to today then report can be empty or incomplete. So please do additional research for this question

Reply:
I didn't find any information about impossibility sync reports in real-time. Google Ads returns data for today

Reply:
@sherifnada We need your opinion here.

1. As for me if we include 'today'  into the current sync means that next sync starts from tomorrow so client will lose data between now and the end of today's day. 

2. as far as I remember from my own experience with Google Ads - it does not provide 'real-time' reports and full data is available only in a few hours .

If we want to include todays report into sync then an alternative solution can be - we should have 1 day overlaps between scans. It guarantees that customers have full reports.

Reply:
Actually, I have found Google docs about it

https://support.google.com/google-ads/answer/2544985?hl=en

- such as clicks, conversions, and impressions) are delayed by less than 3 hours
- conversions attributed using attribution models other than “Last click” are typically delayed up to 15 hours
etc

Reply:
I think it's acceptable to sync data only up to yesterday if there is no other way to guarantee data loss. Ideally, we can pick up a sync exactly where we left off. But if it's not possible, that's acceptable imo. 

What is the technical possibility of doing it that way?

Reply:
@sherifnada we cannot sync data up to today. We use `segments.date` field for filtering data in incremental sync. Google Ads [accepts](https://developers.google.com/google-ads/api/fields/v8/ad_group_ad#segments.date) only date without time for filtering.

Reply:
Ok, lets sync only until yesterday then. we can update that ticket with this information, and mention we can't do it due to limit from api side